### PR TITLE
update single fields

### PR DESCRIPTION
**Pull request by LarryFinn**
Adding ability to just update columns without replacing the rest of 
the columns

### CODE REVIEW

File: src/scala/co/actioniq/luna/dao/DAO.scala [hunk 1]
```diff
@@ -324,6 +324,71 @@ trait DAO[T <: DAOTable.Table[V, I, P], V <: IdModel[I], I <: IdType, P <: JdbcP
...
+  /**
+    * Update three fields without touching the rest of the row.  UpdateActionFunctional actually replaces the whole
+    * row with input mixed with original
+    * @param id id of the object
+    * @param fieldFunction function to go from row to fields
+    * @param setTo tuple value to set fields to
+    * @param validationFieldsOriginal function to validate new field values and original object
+    * @param original original object
+    * @tparam A1 field 1 type
+    * @tparam A2 field 2 type
```
Comment (alexmoore0) [inline src/scala/co/actioniq/luna/dao/DAO.scala:379]: nit: missing `@tparam` for `A3`...

**Review by alexmoore0 — APPROVED** [review]

File: test/scala/co/actioniq/luna/DAOSpec.scala [hunk 2]
```diff
@@ -129,6 +129,36 @@ class DAOSpec extends Specification with Mockito {
       newPlayer.name mustEqual "Mary"
       newPlayer.teamId mustEqual 1L
     }
+    "update two field" in new TestScope with NoopLoggerProvider {
+      val player = awaitResult(playerDao.readByIdFuture(larryId)).get
+      val updateFuture = playerDao.updateFieldFuture[String, Long](
+        larryId,
+        row => (row.name, row.teamId),
```
Comment (danielheyman) [inline test/scala/co/actioniq/luna/DAOSpec.scala:136]: Having more than one seems dangerous as it assumes that you're putting it in the right order. Would it be possible to use a map instead? `(row.name -> "boooo", row.teamId -> 2L)`?

Comment (danielheyman) [reply test/scala/co/actioniq/luna/DAOSpec.scala]: not sure how possible typing this would be though

Comment (LarryFinn) [reply test/scala/co/actioniq/luna/DAOSpec.scala:136]: how i implemented it is similar to how it works in slick.
it works like
```
query
  .map(f => field1, field2)
  .update(value1, value2)
```
  i could do something like
```
def updateBlahblah(
  field1Function: (T) => Rep[A],
  field1Value: A,
  field2Function: (T) => Rep[A1].
  field2Value: A1
)
```
but it's kind of verbose.  i dont think a map would work tho, not sure

**Pull request merged**